src/concrete_solve.jl

```diff
@@ -412,16 +415,61 @@ function DiffEqBase._concrete_solve_adjoint(
         Base.diff_names(Base._nt_names(values(kwargs)),
         (:callback_adj, :callback))}(values(kwargs))
     isq = sensealg isa QuadratureAdjoint
+    kwargs_init = kwargs_adj[Base.diff_names(Base._nt_names(kwargs_adj), (:initializealg,))]
+
+    if haskey(kwargs, :initializealg) || haskey(prob.kwargs, :initializealg)
+        initializealg = haskey(kwargs, :initializealg) ? kwargs[:initializealg] : prob.kwargs[:initializealg]
+    else
+        initializealg = DefaultInit()
+    end
+
+    default_inits = Union{OverrideInit, Nothing, DefaultInit}
+    igs, new_u0, new_p, new_initializealg = if (SciMLBase.has_initialization_data(_prob.f) && initializealg isa default_inits) || initializealg isa CheckInit
```

Comment:
```suggestion
    igs, new_u0, new_p, new_initializealg = if (SciMLBase.has_initialization_data(_prob.f) && initializealg isa default_inits)
```

I don't see why `CheckInit` would take this branch: if CheckInit then you effectively "do nothing" (after validating it's correct to do nothing)

Reply:
But if we must do something, then, does it run the equivalent of OverrideInit?

Reply:
@AayushSabharwal there isn't a custom CheckInit path right?

Reply:
https://github.com/SciML/SciMLBase.jl/blob/master/src/initialization.jl#L192 there is, but the derivative is still guaranteed to be zero.